### PR TITLE
Add direct and serial options to disk devices

### PR DESCRIPTION
**Pull request by wokalski**
I only added support for cloud-hypervisor. The options are also supported in qemu (and possibly others) but i am just using cloud hypervisor.

serial is not documented yet, but it's been supported by cloud hypervisor for a year.

One thing to watch out for is the length of serial, it has to be 20 characters (or less, maybe).

Direct does not improve I/O performance per se, it mostly reduces overhead and makes VMs less noisy (O_DIRECT).

Maybe this PR should be closed wihout merging and we should have a manual way (i.e. a blanket "extraOpts" option) to specify hypervisor specific options because the doesn't pattern doesnt seem sustainable.

I am not using linux and I didn't want to spend time on setting up the formatter for nix (I am usually using nixfmt in my flake). So if you would like to fix formatting please rebase and format.

### CODE REVIEW

File: lib/runners/cloud-hypervisor.nix [hunk 1]
```diff
@@ -26,7 +26,11 @@ let
   useVirtiofs = builtins.any ({ proto, ... }: proto == "virtiofs") shares;
 
   # Transform attrs to parameters in form of `key1=value1,key2=value2,[...]`
-  opsMapped = ops: lib.concatStringsSep "," (lib.mapAttrsToList (k: v: "${k}=${v}") ops);
+  opsMapped = ops: lib.concatStringsSep "," (
+    lib.mapAttrsToList
+      (k: v: "${k}=${v}")
+      (lib.attrsets.filterAttrs (_: v: v != null) ops)
```
Comment (astro) [inline lib/runners/cloud-hypervisor.nix]: `null` could also mean a key without value which isn't obvious. I actually prefer an immediate `} // lib.optionalAttrs (v != null) { ${k} = v;` in the caller. What do you think?

File: nixos-modules/microvm/options.nix [hunk 2]
```diff
@@ -185,6 +185,18 @@ in
...
+            default = null;
+            description = "User-configured serial number for the disk
+            (Currently only respected if using cloud-hypervisor)";
+          };
+          direct = mkOption {
+            type = nullOr bool;
+            default = null;
+            description = "Whether to set O_DIRECT on the disk.
+            (Currently only respected if using cloud-hypervisor)";
+          };
```
Comment (astro) [inline nixos-modules/microvm/options.nix:199]: Makes sense and I'm going to merge it, but only after someone has checked and maybe implemented support for the other hypervisors. :-)

**Review by astro — APPROVED** [review]

File: lib/runners/cloud-hypervisor.nix (hunk 1 above)
Comment (wokalski) [reply lib/runners/cloud-hypervisor.nix]: Sure. Will fix it.

**Comment by astro** [comment]
@wokalski Could you please review?

**Pull request merged**